src/industry_gui.cpp

```diff
@@ -1494,26 +1530,46 @@ class IndustryDirectoryWindow : public Window {
 		GetAllCargoSuffixes(CARGOSUFFIX_OUT, CST_DIR, i, i->type, indsp, i->produced_cargo, cargo_suffix);
 
 		/* Get industry productions (CargoID, production, suffix, transported) */
-		typedef std::tuple<CargoID, uint16, const char*, uint> CargoInfo;
+		struct CargoInfo {
+			CargoID cargo_id;
+			uint16 production;
+			char *suffix;
+			uint transported;
+		};
 		std::vector<CargoInfo> cargos;
 
 		for (byte j = 0; j < lengthof(i->produced_cargo); j++) {
 			if (i->produced_cargo[j] == CT_INVALID) continue;
-			cargos.emplace_back(i->produced_cargo[j], i->last_month_production[j], cargo_suffix[j].text, ToPercent8(i->last_month_pct_transported[j]));
-		}
+			CargoInfo ci{ i->produced_cargo[j], i->last_month_production[j], cargo_suffix[j].text, ToPercent8(i->last_month_pct_transported[j]) };
+			cargos.emplace_back(ci);
```

Comment:
```suggestion
			cargos.emplace_back(i->produced_cargo[j], i->last_month_production[j], cargo_suffix[j].text, ToPercent8(i->last_month_pct_transported[j]));
```
so that the object is actually constructed inplace, or `push_back` instead